File: aws-globalaccelerator-accelerator/src/main/java/software/amazon/globalaccelerator/accelerator/CallbackContext.java
package software.amazon.globalaccelerator.accelerator;

import com.amazonaws.services.globalaccelerator.model.Accelerator;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Builder
@AllArgsConstructor
public class CallbackContext {
    /**
     * The number of attempts remaining before we timeout the stabilization wait
     */
    private Integer stabilizationRetriesRemaining;

    /**
     * Indicates that primary work has completed and we are just waiting
     * for the accelerator to become deployed
     */
    private boolean pendingStabilization;
}

File: aws-globalaccelerator-accelerator/src/main/kotlin/software/amazon/globalaccelerator/accelerator/CallbackContext.kt
package software.amazon.globalaccelerator.accelerator

/**
 * Holds state to be passed between callbacks.
 * stabilizationRetriesRemaining: Number of seconds remaining before giving up on retries.
 * pendingStabilization: Waiting for termination or completion
 */
data class CallbackContext(val stabilizationRetriesRemaining: Int = 0, val pendingStabilization: Boolean = false)
